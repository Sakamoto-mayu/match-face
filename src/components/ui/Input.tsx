'use client';

type Props = React.ComponentProps<'input'>

const Input = (props:Props) => {
  return (
    <div>
      <input
        {...props}
        className={'border border-black'}
      />
    </div>
  );
}

export default Input
