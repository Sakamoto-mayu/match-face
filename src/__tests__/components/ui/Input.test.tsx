import Input from '@/components/ui/Input';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

describe('Input.test.tsx', () => {
  test('スナップショット', () => {
    const { container } = render(<Input id="" />);
    expect(container).toMatchSnapshot();
  });

  test('コールバック関数が反応するかどうか', async () => {
    const onChange = jest.fn();
    render(<Input id="" onChange={onChange} />);
    await userEvent.type(screen.getByRole('textbox'), 'password');
    // userEventは文字を入力する度に発火する→3文字なら3回
    expect(onChange).toHaveBeenCalledTimes(8);
  });
});
