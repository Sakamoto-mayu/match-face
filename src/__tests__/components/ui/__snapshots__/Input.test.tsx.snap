// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input.test.tsx スナップショット 1`] = `
<div>
  <div>
    <input
      class="border border-black"
      id=""
    />
  </div>
</div>
`;
